lib/AST/ModuleNameLookup.cpp

```diff
@@ -90,20 +95,111 @@ static bool updateOverloadSet(NamedCanTypeSet &overloads,
   return true;
 }
 
-/// After finding decls by name lookup, filter based on the given
-/// resolution kind and existing overload set and add them to \p results.
+/// Returns whether it's okay to stop recursively searching imports, given that
+/// we found something, are using a CanTypeSet to resolve overloads, and have
+/// the given \p resolutionKind.
+static bool canReturnEarly(const CanTypeSet &overloads,
+                           ResolutionKind resolutionKind) {
+  switch (resolutionKind) {
+  case ResolutionKind::Overloadable:
+    return false;
+  case ResolutionKind::Exact:
+  case ResolutionKind::TypesOnly:
+    return true;
+  }
+}
+
+/// Returns whether it's okay to stop recursively searching imports, given that
+/// we found something, are using a NamedCanTypeSet to resolve overloads, and
+/// have the given \p resolutionKind.
+static bool canReturnEarly(const NamedCanTypeSet &overloads,
+                           ResolutionKind resolutionKind) {
+  return false;
+}
+
+namespace {
+using LookupCallbackTy = llvm::function_ref<
```

Comment:
At this point in reading the code, I'm wondering "Why do we need a callback for lookups?"